src/Dapr.Actors/Runtime/DaprStateProvider.cs

```diff
@@ -77,10 +80,56 @@ public async Task<bool> ContainsStateAsync(string actorType, string actorId, str
 
         public async Task SaveStateAsync(string actorType, string actorId, IReadOnlyCollection<ActorStateChange> stateChanges, CancellationToken cancellationToken = default)
         {
-            await this.DoStateChangesTransactionallyAsync(actorType, actorId, stateChanges, cancellationToken);
+            await this.daprInteractor.DoStateChangesTransactionallyAsync(this, actorType, actorId, stateChanges, cancellationToken);
         }
 
-        private async Task DoStateChangesTransactionallyAsync(string actorType, string actorId, IReadOnlyCollection<ActorStateChange> stateChanges, CancellationToken cancellationToken = default)
+        public async Task DoStateChangesTransactionallyAsyncGrpc(string actorType, string actorId, IReadOnlyCollection<ActorStateChange> stateChanges, CancellationToken cancellationToken = default)
```

Comment:
Moving this logic into the `daprGrpcInteractor` would remove the need to distinguish between http/grpc right?